tools/mksurfdata_map/src/mkpctPftTypeMod.F90

```diff
@@ -57,7 +57,7 @@ module mkpctPftTypeMod
   end interface pct_pft_type
 
   ! !PRIVATE TYPES:
-  real(r8), parameter :: tol = 1.e-12_r8  ! tolerance for checking equality
+  real(r8), parameter :: tol = 1.e-9_r8  ! tolerance for checking equality
```

Comment:
I had to raise the tolerance for this test to pass:
`convert_from_p2g ERROR: default_pct_p2l must sum to 100`
`sum(default_pct_p2l) =   100.000000000149`

Reply:
This, together with the still-slightly-larger-than-expected differences from before, is making me nervous. This amounts to a relative error of 1e-12, which is about 4 orders of magnitude greater than double precision roundoff.

I feel like we should either (1) trace this greater error back to the source and convince ourselves that this is an acceptable level of error, or (2) play with the source-masking remapping algorithm to try to reduce roundoff-level errors.

Regarding (2), the first thing I'd try is to change this:

```Fortran
          dst_array(no) = dst_array(no) + wt*mask_src(ni)*src_array(ni)/wtnorm(no)
```

I'd delete the division by wtnorm(no) here. Instead, I'd change the where block at the end of the routine:

```Fortran
    where (wtnorm == 0._r8)
       dst_array = nodata
    end where
```

to:

```
    where (wtnorm == 0._r8)
       dst_array = nodata
    elsewhere
       dst_array = dst_array / wtnorm
    end where
```

My gut feeling is that just doing this division once will be less prone to introducing errors in the final result, though I'm not sure of this.

Reply:
Also, if it would help, I'm happy to help with this investigation if you give me directions for how to reproduce what you've been doing for testing.

Reply:
@billsacks I tried your suggestion, but it did not help. I have now pushed new changes that correct the problem.

Reply:
your suggestion (2) above